src/server/routes/parcelDelivery.routes.spec.ts

```diff
@@ -175,4 +181,73 @@ describe('Parcel delivery route', () => {
       data_base64: SERVICE_MESSAGE_CONTENT.toString('base64'),
     });
   });
+
+  describe('Incoming PDA', () => {
+    const pdaContentType = 'application/vnd+relaycorp.awala.pda-path';
+
+    test('Valid PDA should be stored', async () => {
+      const certificatePath = await generatePDACertificationPath(KEY_PAIR_SET);
+      const pdaPath = new CertificationPath(certificatePath.pdaGrantee, [
+        certificatePath.privateEndpoint,
+        certificatePath.privateGateway,
+      ]);
+      const peerConnectionParams = new PrivateEndpointConnParams(
+        PEER_KEY_PAIR.privateGateway.publicKey,
+        PEER_ADDRESS,
+        pdaPath,
+      );
+      const { parcelSerialized } = await generateParcel(
+        parcelRecipient,
+        KEY_PAIR_SET,
+        new Date(),
+        sessionKey,
+        pdaContentType,
+        Buffer.from(await peerConnectionParams.serialize()),
+      );
+      const spyOnSavePeerEndpointChannel = jest.spyOn(
+        server.activeEndpoint,
+        'savePeerEndpointChannel',
+      );
+
+      const response = await server.inject({
+        ...validRequestOptions,
+        payload: parcelSerialized,
+      });
+
+      expect(spyOnSavePeerEndpointChannel).toHaveBeenCalledOnceWith(
+        expect.objectContaining({
+          internetGatewayAddress: peerConnectionParams.internetGatewayAddress,
+        }),
+        server.mongoose,
+      );
```

Comment:
This is not properly tested since it's a different instance of that class that is passed to the actual method. Shall I use method.calls[0] to get the arguments and compare serialized versions?

Reply:
Remember that when we need to do a "power comparison" like that, we should use the `expect.toSatisfy()` matcher from `jest-extended`. For example:

```
expect.toSatisfy<PrivateEndpointConnectionParams>(params => (
  Buffer.from(params.serialize()).equals(Buffer.from(peerConnectionParams.internetGatewayAddress))
))
```

Reply:
Right, then go for it 👍🏾

Reply:
Use `method.calls[0]`.